### PR TITLE
allow ascii data reference profile material model to output seismic v…

### PR DESCRIPTION
**Pull request by jdannberg**
…elocities

### CODE REVIEW

File: include/aspect/material_model/ascii_reference_profile.h [hunk 1]
```diff
@@ -107,6 +107,11 @@ namespace aspect
          * @}
          */
 
+        virtual
```
Comment (tjhei) [inline include/aspect/material_model/ascii_reference_profile.h]: add a comment "add named outputs for seismic properties"

File: source/material_model/ascii_reference_profile.cc [hunk 2]
```diff
@@ -46,6 +50,22 @@ namespace aspect
...
+          // the strings have to be all lower case for the lookup class to find them
+          seismic_vp_index = profile.get_column_index_from_name("seismic_vp");
+          seismic_vs_index = profile.get_column_index_from_name("seismic_vs");
+          seismic_dvp_dT_index = profile.get_column_index_from_name("seismic_dvp_dt");
+          seismic_dvs_dT_index = profile.get_column_index_from_name("seismic_dvs_dt");
+        }
+      catch (...)
+        {
+          // keep the invalid unsigned int entry if the columns do not exist,
+          // we'll handle this below when we fill the outputs
```
Comment (tjhei) [inline source/material_model/ascii_reference_profile.cc]: so what if some of the columns exist? I guess you would only see them if you find them in the correct order. Hrm, not sure if that is important.

File: tests/burnman_seismic_test.prm [hunk 3]
```diff
@@ -0,0 +1,150 @@
+# A cookbook for using seismic velocity data
```
Comment (tjhei) [inline tests/burnman_seismic_test.prm]: update to say what it is based on and that you test seismic output

File: tests/burnman_seismic_test/screen-output [hunk 4]
```diff
@@ -0,0 +1,110 @@
+-----------------------------------------------------------------------------
+-----------------------------------------------------------------------------
+
+   Ascii data file coordinates are not equidistant.
```
Comment (tjhei) [inline tests/burnman_seismic_test/screen-output]: this is kind of ugly. :-(

File: source/material_model/ascii_reference_profile.cc (hunk 2 above)
Comment (jdannberg) [inline source/material_model/ascii_reference_profile.cc]: I added a new function to utilities that returns an invalid unsigned int if the column does not exist (individually for every column name I use it with).

**Comment by jdannberg** [comment]
I think I addressed all of you comments, even the one with the ugly warning :smiley:

**Comment by gassmoeller** [comment]
@tjhei: Your time to press the merge button :smile:

**Comment by gassmoeller** [comment]
@tjhei: Your time to press the merge button :smile:

**Pull request merged**